numpy/core/src/multiarray/compiled_base.c

```diff
@@ -116,10 +116,10 @@ arr_bincount(PyObject *NPY_UNUSED(self), PyObject *args, PyObject *kwds)
     }
     else {
         minlength = PyArray_PyIntAsIntp(mlength);
```

Comment:
Would be great to add a deprecation warning for passing `None` here, so that down the line we have an option to remove it